src/document.ts

```diff
@@ -91,14 +100,14 @@ export class Document {
   }
 
   get view() {
-    if (!this.fields) {
-      return null;
+    // Allows `$view: ~` which indicates the document's body is used as the view.
+    if (this.fields && this.fields['$view'] !== undefined) {
```

Comment:
Is there a reason to compare it to `undefined` instead of just leaving it as a truthy statement?

Reply:
Oh yes. It's because using `$view: ~` (i.e. view is null) has meaning in this proposal. Basically, there may be instances where we want a document to be its own template/view. This feature isn't really supported in Grow, where everything needs a corresponding file in `/views/` for a page to build -- but this is a popular construct in other CMS.

Basically, there may be instances where you want to have a document self-render, i.e. a  customized sitemap feed:

```
---
# In file /content/pages/sitemap-custom.xml.njk
$path: /sitemap-custom.xml
$view: ~
docs:
- !g.doc /content/foo.yaml
- !g.doc /content/baz.yaml
- !g.doc /content/baz.yaml
---
<urlset xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<url>
{% for item in doc.docs %}
<loc>{{item.url}}</loc>
{% endfor %}
</url>
<url>
```

The above syntax means:
- Render my body as the view (`$view: ~`) instead of rendering an external template
- Use the nunjucks renderer (extension = `.njk`)

This is a slight but powerful deviation from Grow that is mostly useful for one-off cases like the above... but I thought the behavior warranted a PR so we could discuss -- after all, it more tightly couples content and presentation which I know is something we generally don't want.

The alternative is to not support self-defined views, and instead require devs to do something like:

```
$view: /views/sitemap-custom.xml.njk
```

And the body would be blank.

Open to feedback either way -- after typing this out, I realized we might want to be more forceful about how people can do this. If you think the proposed `$view: ~` feature isn't useful, I'm totally happy to drop it. :)

Reply:
That is an interesting idea. I am not opposed to the idea of a self-referencing template. Since we have the data in the front matter and then the view would be in the body. It would be handy in certain cases, but not something you would want people using a lot since it is better to use a `view/` file if you are using it in more than one place.

Reply:
Yeah, that's exactly it. I'm just not sure that this is a good idea though (even if other generators do it), so I appreciate the extra review.

What do you think about the name? Maybe `$view: ~` is too "magical" and we should have a special keyword for self-referenced templates? Maybe `$view: self`? @stevenle also any thoughts?

Reply:
i like `self` better than `~` here.

Reply:
Thanks... yeah, I like `$view: self` more as it's clearer. I've updated the PR.